src/main/java/hudson/plugins/sshslaves/SSHLauncher.java

```diff
@@ -817,18 +851,34 @@ static byte[] readInputStreamIntoByteArrayAndClose(InputStream inputStream) thro
     private void copySlaveJarUsingSCP(TaskListener listener, String workingDirectory) throws IOException, InterruptedException {
         SCPClient scp = new SCPClient(connection);
         try {
-            // check if the working directory exists
-            if (connection.exec("test -d " + workingDirectory ,listener.getLogger())!=0) {
+            if(isWindows) {
```

Comment:
Could extract the folder check+creation to a method and write a unit test to cover the new feature?